### ISSUE TITLE
Make `eye` signature consistent with NumPy

### ISSUE THREAD

**Issue by tomwhite** [dask/dask#8934]
**What happened**:

Dask's signature for `eye` is

```
dask.array.eye(N, chunks='auto', M=None, k=0, dtype=<class 'float'>)
```

Whereas NumPy's is

```
numpy.eye(N, M=None, k=0, dtype=<class 'float'>, order='C', *, like=None)
```

The position of the `chunks` parameter in Dask means that a call like `eye(3, 2)` means different things in Dask and NumPy.

**What you expected to happen**: Behaviour the same as NumPy.

**Minimal Complete Verifiable Example**:

```python
>>> import numpy as np
>>> np.eye(3, 2).shape
(3, 2)
>>> import dask.array as da
>>> da.eye(3, 2).shape
(3, 3)
```

**Anything else we need to know?**:

**Environment**:

- Dask version: main
- Python version: 3.8
- Operating System: mac
- Install method (conda, pip, source): source

<details>
<summary>Cluster Dump State:</summary>

</details>

**Comment by tomwhite** [comment]
This would unfortunately be a breaking change. However, we could start by issuing a warning. For example:

```python
>>> import dask.array as da
>>> da.eye(3, chunks=2).shape
(3, 3)
>>> da.eye(3, 2).shape
/Users/tom/projects-workspace/dask/dask/array/creation.py:544: FutureWarning: The `chunks` parameter to `eye` should not be specified by position. Please use `chunks=` instead, since the positional order will be changed in a future release to match NumPy.
  warnings.warn(
(3, 3)
```

See https://github.com/tomwhite/dask/commit/5738f8d041a89a8f617a742acd27777bde0db197

**Comment by tomwhite** [comment]
Any thoughts from @dask/array maintainers on the best course of action here?

**Comment by pentschev** [comment]
I agree, it doesn't seem we have much choice besides a warning for a future breaking change.

This would be a broader topic, but perhaps considering making Dask arguments such as `chunks=` [keyword-only](https://peps.python.org/pep-3102/) may be worth it, like we've done with the `like=` argument in NumPy. Such a change would help avoiding similar discrepancies between Dask and the other PyData APIs, plus I suspect few people would use `chunks=` as a positional argument today, but I may of course be wrong.

**Comment by hmaarrfk** [comment]
I think before undergoing this, I think a valid questions is whether or not numpy API compatibility is desirable.

It is conceivable that the original API writers thought that `chunks` was the most important second parameter.

Now this doesn't work when people want to use `dask.array` as a drop in replacement for numpy, but I think it will rarely work.

That said, if the policy is to make `chunks` a keyword argument only, everywhere, I think that it is reasonable.

I don't think you need to do an "all in one rollout", but I do think it is valuable to have a policy saying"we will accept any PR that makes `chunks` a keyword only argument".

Whatever you choose, providing warnings to the users is useful. 

A while back I designed decorators that would enable the design of the deprecation policy and the warnings for you at the time of writing, you could leave them in at infinitum, or you could  remove them during a cursory cleanup.

They also ensure to pass on the warning to the correct stacklevel, something that seems to always be forgotten:
https://github.com/hmaarrfk/wabisabi/blob/master/wabisabi/kwonly_change.py#L24

scikit-image decided not to use them (but essentially rewrote the same idea) when they decided to have larger changes in their API.

**Comment by tomwhite** [comment]
> perhaps considering making Dask arguments such as `chunks=` [keyword-only](https://peps.python.org/pep-3102/) may be worth it

That would be a good goal to have I think.

> A while back I designed decorators that would enable the design of the deprecation policy and the warnings for you at the time of writing, you could leave them in at infinitum, or you could remove them during a cursory cleanup.
> 
> They also ensure to pass on the warning to the correct stacklevel, something that seems to always be forgotten: https://github.com/hmaarrfk/wabisabi/blob/master/wabisabi/kwonly_change.py#L24

Thanks for sharing!
